### DISCUSSION TITLE
Option to allow servers to use their own sound cards

### DISCUSSION THREAD

**Discussion by kcoul** [apohl79/audiogridder#775] [Ideas]
Hello!

Found this recently and realized that some setups in traditional studios with large format mixing consoles may benefit from the option of not having audio routed over the network. If this feature were available, for example, servers could act similarly to analogue FX racks, or ultra-low-latency hardware synths, allowing analogue mixing/summing, and without the need for audio-over-IP and digital summing unless desired. 

Instruments would instead be able to define a hardware output on a CoreAudio/ASIO sound card available on the server, and FXs could define both a hardware input and output. 

IIUC the internal architecture, FX chains, and Instruments connecting to FX on the same server would work as before, with audio routed directly from one to the next. Only the first FX in each chain and the last FX/instrument would need to define a hardware input/output, respectively.

As a JUCE developer, I would be interested in contributing this feature. Just trying to get a sense of whether others might be interested in this feature!

I realize this feature might cause clocking issues between sound cards, but if they are all synced with word clock, this should not be an issue.

**Comment by apohl79** [comment]
I like the idea! You can mail me at audiogridder (at) e47.org, and we can discuss this further also I can get you up to speed on the dev setup.
